File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        """
        [3, -2, 1, -5, 2, 4]
                   p
                n
        """
        pos = []
        neg = []
        final = []
        for num in nums:
            if num > 0:
                pos.append(num)
            else:
                neg.append(num)
        
        for i in range(len(pos)):
            final.append(pos[i])
            final.append(neg[i])
        return final

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> result(nums.size(), 0);
        int p = 0;
        int n = 1;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] > 0){
                result[p] = nums[i];
                p = p + 2;
            }else{
                result[n] = nums[i];
                n = n + 2;
            }
}
        return result;
    }
};